module2/success/learning_goals.md

```diff
@@ -53,71 +51,80 @@ The expected mastery level can be understood with the following scale:
   * Join Table
 * Write migrations to create tables and relationships between tables
 
-#### Testing & Debugging
+### Testing & Debugging
 
-* Write feature tests utilizing:
-  * RSpec and Capybara
-  * CSS selectors to target specific areas of a page
-  * Sad Path Testing
-* Write model tests with RSpec including validations, and class and instance methods
+* Write feature tests in RSpec that:
+    * Use Capybara to mimic the behavior of users
+    * Use CSS selectors to target specific elements of a page
+    * Test for Sad Paths
+* Write model tests in RSpec that:
+    * Use Shouldamatchers to test relationships and validations
+    * fully test custom-built model methods including edge cases
 * Use Pry or Byebug in Rails files to get more information about an error
-* Use `save_and_open_page` to view the HTML generated when visiting a path in a feature test
-* Utilize the Rails console as a tool to get more information about the current state of a development database
-* Use `rails routes` to get additional information about the routes that exist in a Rails application
+* Use `save_and_open_page` and Chrome Developer Tools to debug HTML
 
-# Functional
+### Web Applications
 
-#### Web Applications
+* Describe and implement ReSTful routing
+* Identify the different components of URLs (protocol, domain, path, query params)
+* Deploy an application to Heroku
 
-* Explain how Cookies/Sessions are used to create and maintain application state
+### Project Management
 
-#### Rails Application Development
+* Use GitHub Projects to create and track User Stories
 
-* Implement partials to break a page into reusable components
-* Make use of flash messages
-* Use Sessions to store information about a user and implement login/logout functionality
-* Use filters (e.g. `before_action`) in a Rails controller
-* Use Inheritance from ApplicationController or a student created controller to store methods that will be used in multiple controllers
-* Limit functionality to authorized users
-* Use BCrypt to hash user passwords before storing in the database
-* Use POROs to logically organize code for objects not stored in the database
+# Functional
 
-#### ActiveRecord
+### Web Applications
 
-* Use built-in ActiveRecord methods to join multiple tables of data, calculate statistics and build collections of data grouped by one or more attributes
+* Describe the HTTP request/response cycle
```

Comment:
I like moving this here. We never test them on this, but we do expect them to understand it.